### PR TITLE
fix pylint errors

### PR DESCRIPTION
**Pull request by hjzccc**

### CODE REVIEW

File: ablator/main/mp.py [hunk 1]
```diff
@@ -321,13 +333,13 @@ def kill_idle(self):
         """
         Kill any ray processes that are idle.
         """
-        p = subprocess.Popen(
+        with subprocess.Popen(
```
Comment (fostiropoulos) [inline ablator/main/mp.py]: this is a bug.

**Review by fostiropoulos — CHANGES_REQUESTED** [review]
I believe the ray sub-process is a bug. Because it will wait inside a context. Have you tested the change that it behaves the same as before? 

Also the codecov bot is failing. Can we fix the failing check or find a way to correctly configure code-cov to not trigger false-positives?

**Comment by MrLYG** [comment]
> I believe the ray sub-process is a bug. Because it will wait inside a context. Have you tested the change that it behaves the same as before?
> 
> Also the codecov bot is failing. Can we fix the failing check or find a way to correctly configure code-cov to not trigger false-positives?



Ok, Let me clarify why code-cov will trigger those failures.
In our project, there are two checks from code-cov:
- codecov/project: verifying the overall test coverage percentage of your entire project. If a commit or pull request decreases the total coverage below a certain threshold, this check will fail.

- codecov/patch: ensuring that the changes introduced in a specific commit or pull request are adequately covered by tests. If the new or updated code isn't sufficiently tested, this check will fail.

Whether need to close the two checks is decided by you @fostiropoulos. If you want close them, Just tell me, which one you want to close or close both.

File: ablator/main/mp.py (hunk 1 above)
Comment (hjzccc) [reply ablator/main/mp.py]: I undo the change here

**Comment by fostiropoulos** [comment]
> > I believe the ray sub-process is a bug. Because it will wait inside a context. Have you tested the change that it behaves the same as before?
> > Also the codecov bot is failing. Can we fix the failing check or find a way to correctly configure code-cov to not trigger false-positives?
> 
> Ok, Let me clarify why code-cov will trigger those failures. In our project, there are two checks from code-cov:
> 
> * codecov/project: verifying the overall test coverage percentage of your entire project. If a commit or pull request decreases the total coverage below a certain threshold, this check will fail.
> * codecov/patch: ensuring that the changes introduced in a specific commit or pull request are adequately covered by tests. If the new or updated code isn't sufficiently tested, this check will fail.
> 
> Whether need to close the two checks is decided by you @fostiropoulos. If you want close them, Just tell me, which one you want to close or close both.

Can we adjust the thresholds to something more reasonable for our project? They seem to be triggered by any difference > 0. Maybe we can set it to a small percentage like 0.5% or similar (maybe based on research on what other projects are using?)

**Pull request merged**